Geany integration

wesley-dean-flexion opened the pull request:
<!--
If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

### What I did
I added a section to the integration documentation that describes how to include hadolint as a linter into Geany's build system.

### How I did it
I copied and pasted the relevant code from configuration.  Specifically, the code kicks off a Docker container with hadolint/hadolint then massages the output with sed into a format that Geany interprets as an error location.  If the output includes a string (in this case, ':WARNING:') then the build (lint) is marked as having failed.

Geany's build system runs the command via `/bin/sh -c ' (stuff) '` with quotes escaped and the following replacements:

%d directory to the file being edited / run
%f filename (basename -- no directory)

Screenshot:
![Screenshot_2019-11-06_15-44-42](https://user-images.githubusercontent.com/45051395/68337050-af620400-00ad-11ea-963c-32fcc21dca51.png)

### How to verify it
Open a Dockerfile, include the code in Geany's build actions, and run it.

SuperSandro2000 commented on docs/INTEGRATION.md:
```diff
@@ -219,3 +219,13 @@ There is an integration [vscode-hadolint][] with [VS Code][], built by [ExiaSR][
+
+> Geany is a powerful, stable and lightweight programmer's text editor that provides tons of useful features without bogging down your workflow. It runs on Linux, Windows and MacOS is translated into over 40 languages, and has built-in support for more than 50 programming languages.
+
+The following can be used as a [build action](https://www.geany.org/manual/current/index.html#build-menu-commands-dialog) t [lint](https://www.geany.org/manual/current/index.html#lint) Dockerfiles.
```
Lonely t

wesley-dean-flexion replied:
fixed

zemanlx commented:
Thank you for your contribution!

Can you please solve linting issues of Markdown?

```
docs/INTEGRATION.md
219   209:1-209:60  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
220  210:1-210:124  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
221   211:1-211:38  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
222   212:1-212:25  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
223   213:1-213:92  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
224   214:1-214:44  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
225   215:1-215:38  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
226   216:1-216:56  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
227   217:1-217:35  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
228   218:1-218:87  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
229   219:1-219:80  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
230   220:1-220:42  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
231   221:1-221:36  warning  Move definitions to the end of the file (after the node at line `229`)  final-definition     remark-lint
232        225:284  warning  Line must be at most 100 characters                                     maximum-line-length  remark-lint
233    229:1-231:4  warning  Missing code-language flag                                              fenced-code-flag     remark-lint
```

Feel free to ask for help if you do not know how to handle them.

wesley-dean-flexion commented:
I've cleaned up the markdown issues in the section I contributed.  I'm using a different markdown linter ( https://github.com/markdownlint/markdownlint ) and it's identified a few issues outside of the content I contributed:

```sh
docker run -it --rm -v $(pwd):/data cytopia/awesome-ci syntax-markdown --path=/data/untitled.md
$ find /data/untitled.md -type f    -print0 | xargs -0 -P 8 -n1  sh -c 'if [ -f "${1}" ]; then mdl --no-warnings "$1" 2>&1; fi' --
/data/untitled.md:23: MD013 Line length
/data/untitled.md:70: MD013 Line length
/data/untitled.md:81: MD013 Line length
/data/untitled.md:195: MD030 Spaces after list markers
/data/untitled.md:197: MD030 Spaces after list markers
/data/untitled.md:158: MD031 Fenced code blocks should be surrounded by blank lines

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

zemanlx commented:
@wesley-dean-flexion Thanks. BTW if you want to see if it will pass our CI linter you can run it locally from hadolint root folder

```shell
docker run --rm -i -v $PWD:/lint/input:ro zemanlx/remark-lint:0.1.4 --frail .
```

wesley-dean-flexion commented:
> @wesley-dean-flexion Thanks. BTW if you want to see if it will pass our CI linter you can run it locally from hadolint root folder
> 
> ```shell
> docker run --rm -i -v $PWD:/lint/input:ro zemanlx/remark-lint:0.1.4 --frail .
> ```

Awesome.  Thanks!

wesley-dean-flexion commented:
I've run remark-lint and cleaned up the warnings with the section I added.  That said, remark-lint noted 6 other line length warnings from other sections of the document:

```
   34:83  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
   37:83  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
   70:94  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
   81:92  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
  112:85  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
  205:83  warning  Line must be at most 80 characters    maximum-line-length  remark-lint
```

These should probably be cleaned up as well.

zemanlx commented:
@wesley-dean-flexion Thank you for fixing those issues. The other ones you have mentioned should not show up if you running it from root folder as it contains configuration in [.remarkrc.yaml](https://github.com/hadolint/hadolint/blob/master/.remarkrc.yaml) which sets line length limit to 100 characters.

Thank you again for your contribution :tada:
